tests/functionnal/Ticket.php

```diff
@@ -1599,4 +1599,92 @@ public function testPrepareInputForAdd($input, $expected, $rich) {
+
+      // simulate "associate myself" feature
+      $ticket_user = new \Ticket_User();
+      $input_ticket_user = [
+         'tickets_id'       => $tickets_id,
+         'users_id'         => \Session::getLoginUserID(),
+         'use_notification' => 1,
+         'type'             => \CommonITILActor::ASSIGN
+      ];
+      $this->integer((int) $ticket_user->add($input_ticket_user))->isGreaterThan(0);
+      $this->boolean($ticket_user->getFromDB($ticket_user->getId()))->isTrue();
+
+      // check status (should be ASSIGNED)
+      $this->boolean($ticket->getFromDB($tickets_id))->isTrue();
+      $this->integer((int) $ticket->fields['status'])
+           ->isEqualto(\CommonITILObject::ASSIGNED);
+
+      // remove associated user
+      $ticket_user->delete([
+         'id' => $ticket_user->getId()
+      ]);
+
+      // check status (should be INCOMING)
+      $this->boolean($ticket->getFromDB($tickets_id))->isTrue();
+      $this->integer((int) $ticket->fields['status'])
+           ->isEqualto(\CommonITILObject::INCOMING);
+
+      // drop UPDATE right to TU_USER and redo "associate myself"
+      $saverights = $_SESSION['glpiactiveprofile'];
+      $_SESSION['glpiactiveprofile']['ticket'] -= \UPDATE;
+      $this->integer((int) $ticket_user->add($input_ticket_user))->isGreaterThan(0);
+      $this->boolean($ticket_user->getFromDB($ticket_user->getId()))->isTrue();
+
+      // check status (should be ASSIGNED)
+      $this->boolean($ticket->getFromDB($tickets_id))->isTrue();
+      $this->integer((int) $ticket->fields['status'])
+           ->isEqualto(\CommonITILObject::ASSIGNED);
+
+      // remove associated user
+      $ticket_user->delete([
+         'id' => $ticket_user->getId()
+      ]);
+
+      // check status (should be INCOMING)
+      $this->boolean($ticket->getFromDB($tickets_id))->isTrue();
+      $this->integer((int) $ticket->fields['status'])
+           ->isEqualto(\CommonITILObject::INCOMING);
+
+      // remove associated user
+      $ticket_user->delete([
+         'id' => $ticket_user->getId()
+      ]);
+
+      // check with very limited rights and redo "associate myself"
+      $_SESSION['glpiactiveprofile']['ticket'] = \CREATE
+                                               + \Ticket::READMY;
+                                               + \Ticket::READALL;
+                                               + \Ticket::READGROUP;
+                                               + \Ticket::OWN; // OWN right must allow self-assign
+      $this->integer((int) $ticket_user->add($input_ticket_user))->isGreaterThan(0);
+      $this->boolean($ticket_user->getFromDB($ticket_user->getId()))->isTrue();
+
+      // check status (should still be ASSIGNED)
+      $this->boolean($ticket->getFromDB($tickets_id))->isTrue();
+      $this->integer((int) $ticket->fields['status'])
+           ->isEqualto(\CommonITILObject::ASSIGNED);
+
+      // restore right
+      $_SESSION['glpiactiveprofile'] = $saverights;
```

Comment:
would be better I guess to reset right just after `add()` has been called; to prevent issues when following tests fails.